lib/Basics/process-utils.cpp

```diff
@@ -966,6 +966,36 @@ void TRI_CreateExternalProcess(char const* executable,
 /// @brief Reads from the pipe of processes
 ////////////////////////////////////////////////////////////////////////////////
 
+void TRI_ClosePipe(ExternalProcess* process,
+                   bool read) {
+  if (process == nullptr ||
+      (read && TRI_IS_INVALID_PIPE(process->_readPipe)) ||
+      (!read && TRI_IS_INVALID_PIPE(process->_writePipe))
+      ) {
+    return;
+  }
+
+  auto pipe = (read) ? &process->_readPipe : &process->_writePipe;
+  
+#ifndef _WIN32
+  if (*pipe != -1) {
+    FILE *stream = fdopen(*pipe, "w");
+    fflush(stream);
+    close(*pipe);
```

Comment:
In theory, `close(fd)` can fail, and we are not checking for it.
However, in reality there is not so much that can be done if it fails, except to log an error or such.